src/log.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <iostream>
+#include <sstream>
+
+#include "bpftrace.h"
+
+namespace bpftrace {
+
+// clang-format off
+enum class LogType
+{
+  debug,
+  info,
+  warning,
+  error
+};
+// clang-format on
+
+class Log
+{
+public:
+  static Log& get();
+  void take_input(LogType type,
+                  const location& loc,
+                  bool with_loc,
+                  std::ostream& out,
+                  std::string&& input);
+  inline void set_source(std::string filename, std::string source)
```

Comment:
Nit
```suggestion
  inline void set_source(const std::string& filename, std::string source)
```

Reply:
I guess we may want  `const std::string& source`, too?